mirrord/layer/src/file/ops.rs

```diff
@@ -30,11 +30,18 @@ impl RemoteFile {
 
 impl Drop for RemoteFile {
     fn drop(&mut self) {
-        trace!("dropping RemoteFile {self:?}");
+        // Warning: Don't log from here. This is called when self is removed from OPEN_FILES, so
+        // during the whole execution of this function, OPEN_FILES is locked.
+        // When emitting logs, sometimes a file `write` operation is required, in order for the
+        // operation to complete. The write operation is hooked and at some point tries to lock
+        // `OPEN_FILES`, which means the thread deadlocks with itself (we call
+        // `OPEN_FILES.lock()?.remove()` and then while still locked, `OPEN_FILES.lock()` again)
         let closing_file = Close { fd: self.fd };
 
         if let Err(err) = blocking_send_file_message(FileOperation::Close(closing_file)) {
-            warn!("Failed to send close file {self:?} message: {err:?}");
+            println!(
```

Comment:
Is printing instead of emitting a warning a good enough solution?
I'm not sure that printing is safer than logging, but when I fixed the problematic `trace` above with a `println` the deadlock did not happen (for many tests).
Other alternatives would be not to show any message at all, or just to risk it with the warning log, because the deadlock does not happen every time, and this situation (sending message to main layer thread fails) should already be very rare. Another alternative would be to `expect`, because if we can't send messages to the main thread maybe we should exit anyways.

Reply:
agree with the latter

Reply:
Changed to `expect` https://github.com/metalbear-co/mirrord/pull/1009/commits/c74c2151dbf1a8f643c3594188a6384b84b7946c.